perception/lidar_centerpoint_tvm/lib/centerpoint_tvm.cpp

```diff
@@ -39,135 +42,113 @@ namespace perception
 namespace lidar_centerpoint_tvm
 {
 
-VoxelEncoderPreProcessor::VoxelEncoderPreProcessor(
-  const tvm_utility::pipeline::InferenceEngineTVMConfig & config,
-  const CenterPointConfig & config_mod)
-: max_voxel_size(config.network_inputs[0].second[0]),
-  max_point_in_voxel_size(config.network_inputs[0].second[1]),
-  encoder_in_feature_size(config.network_inputs[0].second[2]),
-  datatype_bytes(config.tvm_dtype_bits / 8),
-  config_detail(config_mod)
-{
-  encoder_in_features.resize(max_voxel_size * max_point_in_voxel_size * encoder_in_feature_size);
-  // Allocate input variable
-  std::vector<int64_t> shape_x{1, max_voxel_size, max_point_in_voxel_size, encoder_in_feature_size};
-  TVMArrayContainer x{
-    shape_x,
-    config.tvm_dtype_code,
-    config.tvm_dtype_bits,
-    config.tvm_dtype_lanes,
-    config.tvm_device_type,
-    config.tvm_device_id};
-  output = x;
-}
-
-TVMArrayContainerVector VoxelEncoderPreProcessor::schedule(const MixedInputs & voxel_inputs)
+TVMScatterIE::TVMScatterIE(
+  tvm_utility::pipeline::InferenceEngineTVMConfig config, const std::string & pkg_name,
+  const std::string & function_name)
+: config_(config)
 {
-  /// generate encoder_in_features from the voxels
-  generateFeatures(
-    voxel_inputs.features, voxel_inputs.num_points_per_voxel, voxel_inputs.coords,
-    voxel_inputs.num_voxels, config_detail, encoder_in_features);
+  std::string network_prefix =
+    ament_index_cpp::get_package_share_directory(pkg_name) + "/models/" + config.network_name + "/";
+  std::string network_module_path = network_prefix + config.network_module_path;
 
-  TVMArrayCopyFromBytes(
-    output.getArray(), encoder_in_features.data(),
-    max_voxel_size * max_point_in_voxel_size * encoder_in_feature_size * datatype_bytes);
+  std::ifstream module(network_module_path);
+  if (!module.good()) {
+    throw std::runtime_error("File " + network_module_path + " specified in config.hpp not found");
+  }
+  module.close();
+  tvm::runtime::Module runtime_mod = tvm::runtime::Module::LoadFromFile(network_module_path);
 
-  return {output};
-}
+  f = runtime_mod.GetFunction(function_name);
```

Comment:
nit: please use something more evocative and with more letters than `f`. `tvm_module`? idk